src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -101,6 +101,13 @@ export class Kernel implements IKernel {
         return this._kernelSocket.asObservable();
     }
     private notebook?: INotebook;
+    /**
+     * If the session died, then ensure the status is set to `dead`.
+     * We need to provide an accurate status.
+     * `unknown` is generally used to indicate jupyter kernel hasn't started.
+     * If a jupyter kernel dies after it has started, then status is set to `dead`.
+     */
+    private isKernelDead?: boolean;
```

Comment:
I want to remove this, horrible to have more states (basically yet another wrapper around `status`).
But we need this for raw kernels.
Once we clean up Kernels as a whole this should go away..

wait, i think i'll remove this.